youtube_dl/extractor/wrzuta_playlist.py

```diff
@@ -0,0 +1,135 @@
+        },
+    }, {
+        'url': 'http://andziaoox.wrzuta.pl/playlista/1TOgrSX1hFR/andziaoox_moja_playlista_1',
+        'playlist_mincount': 29,
+        'info_dict': {
+            'id': '1TOgrSX1hFR',
+            'title': 'andziaoox_moja_playlista_1',
+            'description': 'andziaoox_moja_playlista_1',
+        },
+    }]
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        extracted_id = mobj.group('id')
+        uploader = mobj.group('uploader')
+        media_title = mobj.group('playlist_title')
+
+        return self._extract_playlist_content(
+            url=url,
+            uploader=uploader,
+            playlist_id=extracted_id,
+            playlist_title=media_title,
+        )
+
+    def _extract_playlist_content(self, url, uploader, playlist_id, playlist_title):
+        entries = []
+
+        webpage = self._download_webpage(
+            url_or_request=url,
+            video_id=playlist_id,
+        )
+        playlist_size = self._extract_playlist_size(webpage)
+
+        there_are_entries_for_extraction = playlist_size > 0
+        if there_are_entries_for_extraction:
+            entries = self._extract_entries_from_webpage(webpage)
+
+            number_of_extracted_entries_still_lower_than_list_size = playlist_size > len(entries)
+            if number_of_extracted_entries_still_lower_than_list_size:
+                entries += self._collect_and_extract_remaining_entries(playlist_id, uploader)
+
+        return self.playlist_result(
+            entries=entries,
+            playlist_id=playlist_id,
+            playlist_title=playlist_title,
+            playlist_description=playlist_title,
+        )
+
+    def _collect_and_extract_remaining_entries(self, playlist_id, uploader):
+        playlist_content = self._download_json(
+            url_or_request='http://{uploader_id}.wrzuta.pl/xhr/get_playlist_offset/{playlist_id}'.format(
+                uploader_id=uploader,
+                playlist_id=playlist_id,
+            ),
+            video_id=playlist_id,
+            note='Downloading playlist content as JSON metadata',
+            errnote='Unable to download playlist content as JSON metadata',
+        )
+        entries = [self.url_result(entry['filelink']) for entry in playlist_content['files']]
+        return entries
+
+    def _extract_entries_from_webpage(self, webpage):
+        return map(
+            lambda entry_url: self.url_result(entry_url),
+            self._extract_html_entries(webpage)
+        )
+
+    @staticmethod
+    def _extract_html_entries(webpage):
+        _entry_pattern = r'<a href="(?P<playlist_entry_url>[^"]+)" target="_blank" class="playlist\-file\-page">'
+        return re.findall(_entry_pattern, webpage)
+
+    @staticmethod
+    def _extract_playlist_size(webpage):
+        _size_pattern = r'<div class="playlist-counter">[0-9]+/(?P<playlist_size>[0-9]+)</div>'
+        matched_obj = re.search(_size_pattern, webpage)
```

Comment:
> Use _html_search_regex() instead
For `_extract_html_entries()` might be difficult to replace regexp with `_html_search_regex() `. I had quick look at the method and didn't see a way to return collection (or iterator) of matched groups. I could use `clean_html()` and `strip()` functions along with the regular expression instead of `_extract_html_entries()` function or you could help me by giving a hint of usage - perhaps I missed something.